dotnet/samples/KernelSyntaxExamples/Example77_HandlebarsPromptSyntax.cs

```diff
@@ -0,0 +1,320 @@
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddOpenAIChatCompletion(
+                modelId: openAIModelId,
+                apiKey: openAIApiKey)
+            .Build();
+
+        KernelPlugin productMagicianPlugin = GenerateProductMagicianPlugin();
+        kernel.Plugins.Add(productMagicianPlugin);
+
+        await TestProductFunctionsAsync(kernel);
+
+        await RunHandlebarsPlannerSampleAsync(kernel);
+
+        await RunHandlebarsTemplateSample01Async(kernel);
+
+        await RunHandlebarsTemplateSample02Async(kernel);
+
+        await RunHandlebarsTemplateSample03Async(kernel);
+    }
+
+    private async Task RunHandlebarsTemplateSample03Async(Kernel kernel)
+    {
+        string handlebarsTemplate3 = @"
+            {{!-- example of set with input and function calling with two syntax types --}}
+            {{set ""companyDescription"" input}}
+            {{set ""productNames"" (productMagician-GenerateJSONProducts companyDescription)}}
+
+            {{#if generateEngagingDescriptions}} 
+                {{!-- Step 2: Create array for storing final descriptions --}}
+                {{set ""finalDescriptions"" (array)}}
+
+                {{!-- Step 3: Iterate over each generated product name --}}
+                {{#each productNames}}
+                    {{#each this}}
+                      {{!-- Step 3.1: Concatenating productName to initial company description --}}
+                      {{set ""productDescription"" (concat ""Product Name: "" this.name "" Description: "" this.description)}}
+
+                      {{!-- Step 3.2: Generate compelling description for each productName --}}
+                      {{set ""compellingDescription"" (productMagician-GenerateProductCompellingDescription productDescription)}}
+
+                      {{!-- Step 3.3: Concatenate compelling description and product number --}}                  
+                      {{set ""outputDescription"" (concat ""PRODUCT :"" this.name "" Engaging Description: "" compellingDescription)}}
+
+                      {{!-- Step 3.4: Add output description to the list --}}
+                      {{set ""finalDescriptions"" (array finalDescriptions outputDescription)}}
+                      {{set ""finalDescriptionsV2"" (concat finalDescriptionsV2 "" -- "" outputDescription)}}
+                    {{/each}}
+                {{/each}}
+
+                {{!-- Step 4: Print all product names and compelling descriptions --}}
+                OUTPUT The following product descriptions as is, do not modify anything:
+                {{json finalDescriptionsV2}}
+    
+            {{else}} 
+                {{!-- Example of concatenating text and variables to finally output it with json --}}
+                {{set ""finalOutput"" (concat ""Description 1: "" productNames "" Description 2: "" productNames2)}}
+                {{json finalOutput}}
+            {{/if}}";
```

Comment:
In terms of showcase a HB prompt template, these are much too complex. 

Example of how you can reduce complexity:
```handlebars
{{!-- Omit the initial variable mapping of inputs (this is required for the planner to reduce error) but as a dev executing their own HB template, you should have full context over how input are maps. --}}
{{set ""productNames"" (productMagician-GenerateJSONProducts input)}}

{{#if generateEngagingDescriptions}} 
{{!-- Since SK doesn't support mutable arrays, you must concatenate each result onto a string. --}}
    {{set ""finalProducts"" """"}}

    {{#each productNames}}
{{!-- Refactor your GenerateJSONProducts function to return just an array of products, then remove the inner {{#each}}} block. It's unnecessarily complex. --}}
        {{set ""productDescription"" (concat ""Product Name: "" this.name "" Description: "" this.description)}}
        {{set ""compellingDescription"" (productMagician-GenerateProductCompellingDescription this)}}
        {{set ""outputProduct"" (concat ""PRODUCT :"" this.name "" Engaging Description: "" compellingDescription)}}

{{!-- Check if you're on the first object; if not, concatenate to end of string.  --}}
        {{#if @first}}{{set ""finalProducts"" outputProduct}}
        {{else}}{{set ""finalProducts"" (concat finalProducts ""\n\n"" outputProduct)}}
        {{/if}}
    {{/each}}
{{!--  OUTPUT The following product descriptions as is, do not modify anything --}}   
    {{json finalProducts}}
{{else}} 
{{!-- Return products as is since the conditional is checking whether descriptions should be regenerated. --}}
    {{json productNames}}
{{/if}}
```